packages/nextjs/pages/_app.tsx

```diff
@@ -49,8 +49,14 @@ const ScaffoldEthApp = ({ Component, pageProps }: AppProps) => {
         </div>
         <Toaster />
       </RainbowKitProvider>
-    </WagmiConfig>
+    </>
   );
 };
 
-export default ScaffoldEthApp;
+const ScaffoldEthAppWithWagmi = (props: AppProps) => (
+  <WagmiConfig config={wagmiConfig}>
+    <ScaffoldEthApp {...props} />
+  </WagmiConfig>
+);
+
```

Comment:
Because the `useTargetNetwork` hook (which internally uses wagmi's `useNetwork` hook) is used inside the native price hook, so this must be inside the wagmi context. I might revisit some things in `_app.tsx` to make it a bit cleaner.

Reply:
That makes sense!

We could also have `ScaffoldEthAppWithProviders` where we have all the declared providers.

Reply:
Cleaned it up a little with your suggestion :+1: